src/Http/Request/Body/JsonSchemaValidatingRequestBodyParser.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Http\Request\Body;
+
+use CultuurNet\UDB3\Http\ApiProblem\ApiProblem;
+use CultuurNet\UDB3\Http\ApiProblem\SchemaError;
+use Opis\JsonSchema\Errors\ErrorFormatter;
+use Opis\JsonSchema\Validator;
+use Psr\Http\Message\ServerRequestInterface;
+
+final class JsonSchemaValidatingRequestBodyParser implements RequestBodyParser
+{
+    private const SCHEMA_LOCATIONS = __DIR__ . '/../../../../vendor/publiq/stoplight-docs-uitdatabank/models/';
+    private const MAX_ERRORS = 100;
+
+    private RequestBodyParser $baseParser;
+    private Validator $validator;
+    private string $jsonSchema;
+
+    private function __construct(string $jsonSchema)
+    {
+        $this->jsonSchema = $jsonSchema;
+        $this->baseParser = new JsonRequestBodyParser();
+        $this->validator = new Validator(null, self::MAX_ERRORS);
+    }
+
+    public static function forStoplightModelFileName(string $stoplightModelFileName): self
+    {
+        $stoplightModelFileName = ltrim($stoplightModelFileName, '/');
+        $schema = file_get_contents(self::SCHEMA_LOCATIONS . $stoplightModelFileName);
```

Comment:
I guess the impact of this disk IO will be minimal? I know we should not do any premature performance optimisations but I'm always sceptical if I see any disk IO 😉

Reply:
Should be the same as the disk IO of loading the yaml config files. And it should usually only be one schema per request.

Reply:
Yes, so no need to worry about it 😀